proposals/exception-handling/Exceptions.md

```diff
@@ -288,28 +288,104 @@ end
 If `call $foo` throws, searching for a catching block first finds `delegate`,
 and because it delegates exception handling to catching instructions associated
 with `$l1`, it will be next checked by the outer `catch` and then `catch_all`
-instructions. When the specified label within a `delegate` instruction does not
-correspond to a `try` instruction, it is a validation failure.
+instructions.
+
+`delegate` can also target `catch`-less `try`s or non-`try` block constructs
+like `block`s or `loop`s, in which case the delegated exception is assumed to
+propagate to the outer scope and will be caught by the next matching
+try-catches. In the examples, catches are annotated with `($label_name)` to
+clarify which `try` it belongs to for clarification; it is not the official
+syntax.
+```
+try $l0
+  block $l1
+    try
+      call $foo
+    delegate $l1  ;; delegates to 'catch ($l0)'
+  end
+catch ($l0)
+end
+```
 
-Note that the example below is a validation failure:
+`delegate` can only target `catch`/`catch_all`s that are below the current
+instruction, i.e., it can only delegate downwards. It also targets all
+`catch`/`catch_all`s of a `try` as a whole and does not target individual
+`catch`/`catch_all` within a `try`. Here is another example:
```

Comment:
Nit: Delegate is no longer targeting catch after this change, it simply targets a block. I would perhaps rephrase this accordingly and say that all properties of Wasm's structured control flow labels apply, i.e., you can only target labels of outer blocks.

Reply:
True, but the example below is, the `delegate` targets `$l1`, which is outside of the instruction because the label is attached to the `try`, but because it is below `catch` so it ends up not targeting `$l1` but rather `$l0`, and this was what I tried to explain. Do you think simply describing that `delegate` target labels of outer blocks can explain this case?

Reply:
@rossberg Do you still think this needs fixing?

Reply:
I would say s.th like this:

"""
Like branches, `delegate` can only target outer blocks, and effectively rethrows the exception in that block. Consequently, delegating to a specific `catch` or `catch_all` handler requires targeting the respective label from within the associated `try` block. Delegating to a label from within a `catch` block does delegate the exception to the next enclosing handler -- analogous to performing a `throw` within a `catch` block, that handler is no longer active at that point.
"""

Reply:
Done